### PR TITLE
Topic recognition 

### PR DESCRIPTION
**Pull request by Abhinavmohindersingh**
Super-resolution CNN model to generated high resolution images of the ADNI brain scan dataset with a downsample factor of 4. Following files are included in the project under Brain_MRI_SuperResolution directory.

dataset.py - For the purpose of preprocessing the data-sample  and feeding data to the model.

modules.py - CNN model consisting of residual block and convulational layers

predict.py - Functions used for predicting the low resolution images and display the upscaled images.

train.py - Utilised for training the super resolution model and plotting graphs for training and validation losses and PSNR metrics.

README.md - containing all information about the report

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Files have been implemented in the wrong folder. The student has combined SRCNN with PixelCNN, which results in good outcomes for the output images. However, they only downscale images to 100x100. Also, residual artefacts in the upscaled images are probably a result of the final layer being a ConvTranspose2D layer. Target image size is also 400x400. Overall problem was made easier by their chosen resolutions, but code and README seems ok.

**[Recognition Problem]**
"modules.py":

- Implementation of Super Rez in Tensorflow.
- Adds a residual block before upscaling block.
- Adds a convolutional layer after the first PixelShuffle upsampling block.
- Adds a final layer that is a ConvTranspose2D block
- so upsampling 2x via pixelshuffle then 2x via convtranspose.

"dataset.py":

- Creates some functions to downsample the images and store them to a folder, rather than computing on the fly.
- Creates a function that then loads all of the images into an array and stores them in memory.
- Student includes an image sharpen function, which might help in the reconstruction?

"train.py":

- Loads AD and NC images separately, but then doesn't randomize before splitting into train and validation. It may be possible that either AD or NC are not in the training or validation set.
- Uses checkpoints and early stopping.
- Creates training plots.
- "Target Size" for Low Rez is 100x100 and High Rez is 400x400. Seems the student is resizing both the input and ground truth, rather than just the input to 60x64. This makes the super rez problem easier, as the initial image is only ~2x downscaled.

"predict.py":

- Correctly loads the model. 
- Provides some reconstructions, again starting from the 100x100 image size.
- No metrics computed on the test set.

"README.MD":

- Contains description of approach, however figure used is from SRCNN which doesn't itself use pixelshuffle or ConvTranspose2D.
- Much lower validation scores compared to training.
- Dependencies included.

**[Commit Log]**
Commit log looks ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, but no metrics on test -1
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good missing usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Cannot merge because of changes to repo README and adding DStores files to repo, please remove and revert to allow merge. Does not affect grade.

**Comment by Abhinavmohindersingh** [comment]
Sir, done with all of the deletion and reverted back README.md to original.

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
I think the adjustment is insufficient. All student-constructed documents should be packed in one document under the ‘recognition’ folder, which means you just need to create one folder under the original ./recognition/ folder without modifying any other original files and put all your files into the student-constructed folder. Try to fix it so that Shakes can merge your pull request, though you will not lose any marks.

**Comment by Abhinavmohindersingh** [comment]
Changed and put everything in recognition folder
